### PR TITLE
New announcement

### PR DESCRIPTION
**Pull request by aellispierce**

### CODE REVIEW

**Comment by aellispierce** [comment]
I saw that jquery had a validate plugin that could make this a little fancier but didn't know if it was worth including yet

File: app/assets/new_announcement.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    Title:<br>
+    <input type="text" name="announcement[title]"> <br>
+    Body:<br>
+    <textarea name="announcement[body]"></textarea><br>
+    Interests:<br>
+    <input type="text" name="interest_names"> <br> <br>
+    <input type="submit" value="Submit">
+  </form>
+  <script>
+    app = require('constable-api');
```
Comment (drapergeek) [inline app/assets/new_announcement.html:18]: I would recommend keeping your markup and your JS completely separate.

File: app/constable-api.js [hunk 2]
```diff
@@ -27,13 +27,22 @@ class ConstableApi {
...
+  validateForm(e) {
+    const formData= $("#new").serializeArray();
+    if(formData[0]["value"] == ""){
+      $("body").append('Title is required');
+      e.preventDefault();
+    } else if (formData[1]["value"] == ""){
+      $("body").append('Body is required');
+      e.preventDefault();
+    } else {
+      this.newAnnouncement();
```
Comment (drapergeek) [inline app/constable-api.js:44]: I think renaming this would provide a lot of clarity.

File: app/constable-api.js [hunk 3]
```diff
@@ -27,13 +27,22 @@ class ConstableApi {
...
     });
   }
+  validateForm(e) {
+    const formData= $("#new").serializeArray();
+    if(formData[0]["value"] == ""){
+      $("body").append('Title is required');
+      e.preventDefault();
+    } else if (formData[1]["value"] == ""){
+      $("body").append('Body is required');
+      e.preventDefault();
```
Comment (drapergeek) [inline app/constable-api.js:42]: Is there any reason we can't start off with `e.preventDefault()`?

Comment (aellispierce) [inline app/constable-api.js:42]: I only want it to happen for two of the three conditions

**Pull request closed without merging**